pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -40,17 +44,36 @@ import (
 
 var defaultDestinationRule = networking.DestinationRule{}
 
+var istioMtlsTransportSocketMatch = &structpb.Struct{
+	Fields: map[string]*structpb.Value{
+		model.TLSModeLabelShortname: {Kind: &structpb.Value_StringValue{StringValue: model.IstioMutualTLSModeLabel}},
+	},
+}
+
+// h2UpgradeMap specifies the truth table when upgrade takes place.
+var h2UpgradeMap = map[upgradeTuple]bool{
+	{meshconfig.MeshConfig_DO_NOT_UPGRADE, networking.ConnectionPoolSettings_HTTPSettings_UPGRADE}:        true,
+	{meshconfig.MeshConfig_DO_NOT_UPGRADE, networking.ConnectionPoolSettings_HTTPSettings_DO_NOT_UPGRADE}: false,
+	{meshconfig.MeshConfig_DO_NOT_UPGRADE, networking.ConnectionPoolSettings_HTTPSettings_DEFAULT}:        false,
+	{meshconfig.MeshConfig_UPGRADE, networking.ConnectionPoolSettings_HTTPSettings_UPGRADE}:               true,
+	{meshconfig.MeshConfig_UPGRADE, networking.ConnectionPoolSettings_HTTPSettings_DO_NOT_UPGRADE}:        false,
+	{meshconfig.MeshConfig_UPGRADE, networking.ConnectionPoolSettings_HTTPSettings_DEFAULT}:               true,
+}
+
 // ClusterBuilder interface provides an abstraction for building Envoy Clusters.
 type ClusterBuilder struct {
 	proxy *model.Proxy
 	push  *model.PushContext
+	// httpProtocolOptions stores the HttpProtocolOptions per cluster.
+	httpProtocolOptions map[string]*http.HttpProtocolOptions
```

Comment:
I can change that - Only reason, I did not use that model is we do not need clusters to be stored. But if it helps other uses and aligns with existing model, I can definitely change.

Reply:
I don't think we should have ClusterBuilder store it, that would be a bit confusing to have this "global" map structure I think.  instead we can pass around RawCluster instead of Cluster? similar to how in listener we build up an istio internal structure then at the very last moment do something like finalizeListener().